include/rocksdb/merge_operator.h

```diff
@@ -115,6 +115,18 @@ class MergeOperator : public Customizable {
     // client can set this field to the operand (or existing_value) instead of
     // using new_value.
     Slice& existing_operand;
+    // Indicates a specific failure reason. It is only meaningful to provide a
+    // failure reason when returning `false` from the API populating the
+    // `MergeOperationOutput`. Currently RocksDB operations handle these
+    // `Status` values as follows:
+    //
+    // - `Status::OK()`: fallback to default (`Status::Corruption()`)
+    // - `Status::Corruption()`: any op fails with `Status::Corruption()`
+    // - `Status::Aborted()`:
+    //   - Background ops (flush/compaction) copy the original input operands to
+    //     the output file without any merging
+    //   - User read ops fail with `Status::Aborted()`
```

Comment:
This description mentions how RocksDB would treat those statuses, which is important. It is a return status of an operation, and we might need to explain what the status means for this routine itself. I guess "corruption" means that the merge operator cannot decode one of the operands and believe the data is corrupted? "aborted" means that the data could be decoded but it doesn't know how to merge them? It's better that users can reason it from the data and operator itself. Otherwise, "status" is a very confusing name, and perhaps should be replaced by another enum.

Reply:
Yeah, it could be achieved using an enum. I think the situation is the user wants to control the failure handling (limit the blast radius) irrespective of what went wrong inside the merge operator. For that it sounds like we should use the enum approach. Will try that - thanks.